include/ISceneNode.h

```diff
@@ -276,30 +277,29 @@ namespace scene
 				child->grab();
 				child->remove(); // remove from old parent
 				Children.push_back(child);
+				child->Iterator = Children.end();
+				(*child->Iterator)--;
```

Comment:
It does make sense to me that the iterator never changes, but is this guaranteed by the C++ spec?

Reply:
See "Iterator Invalidation" table here: https://en.cppreference.com/w/cpp/container

Btw., prefer pre-increment (doesn't need to copy the iterator):
```suggestion
				--(*child->Iterator);
```